consensus/src/experimental/commit_phase.rs

```diff
@@ -0,0 +1,327 @@
+                    // cancel out an item from local_cache and an item from local_queue
+                    if ledger_info_occupied_entry
+                        .get()
+                        .check_voting_power(&self.verifier)
+                        .is_ok()
+                    {
+                        commit(
+                            &self.execution_proxy,
+                            &front.vecblocks,
+                            &front.ledger_info_sig,
+                        )
+                        .await
+                        .expect("Failed to commit the executed blocks.");
+                        assert!(
+                            self.committed_round
+                                < front.ledger_info_sig.ledger_info().commit_info().round()
+                        );
+                        self.committed_round =
+                            front.ledger_info_sig.ledger_info().commit_info().round();
+                        self.commit_msg_sender
+                            .send(CommitPhaseMessage::CommitDecision(
+                                ledger_info_occupied_entry.get().clone(),
+                            ))
+                            .await?;
+                        ledger_info_occupied_entry.remove_entry();
+                        mut_local_queue.pop_front();
+                    } else {
+                        break;
+                    }
+                }
+                Entry::Vacant(_) => {
+                    break;
+                }
+            }
+        }
+        Ok(())
+    }
+
+    pub async fn process_executed_blocks(
+        &mut self,
+        vecblock: Vec<ExecutedBlock>,
+        ledger_info: LedgerInfoWithSignatures,
+    ) -> anyhow::Result<()> {
+        let new_ledger_info = LedgerInfo::new(
+            vecblock.last().unwrap().block_info(),
+            ledger_info.ledger_info().consensus_data_hash(),
+        );
+
+        let sig = self
+            .safety_rules
+            .lock()
+            .sign_commit_vote(ledger_info.clone(), new_ledger_info.clone())?;
+        // if fails, it needs to resend, otherwise the liveness might compromise.
+        self.commit_msg_sender
+            .send(CommitPhaseMessage::CommitVote(
+                self.author,
+                new_ledger_info.clone(),
+                sig,
+            ))
+            .await?;
+        // note that this message will also reach the node itself
+
+        self.local_queue
+            .push_back(PendingBlocks::new(vecblock, ledger_info));
+
+        Ok(())
+    }
+
+    pub async fn start(mut self) {
+        loop {
+            select! {
+                (vecblock, ledger_info) = self.commit_channel_recv.select_next_some() => {
+                    map_err_shortcut!(
+                        self.process_executed_blocks(vecblock, ledger_info).await,
+                        "Error in processing executed blocks"
+                    ).ok();
```

Comment:
why do we need the ok for every call?

Reply:
The compiler will issue a warning that I didn't use the result which must be used.

Reply:
oh the idiomatic way of doing this is
if let Err(e) = self.process() {
  ...
}

Reply:
Resolved